tests/rptest/services/redpanda.py

```diff
@@ -278,13 +291,29 @@ def __init__(self,
                  cloud_storage_readreplica_manifest_sync_timeout_ms: Optional[
                      int] = None,
                  bypass_bucket_creation: bool = False):
+        self.cloud_storage_type = CloudStorageType.S3
+        if hasattr(test_context, 'injected_args') \
+        and test_context.injected_args is not None \
+        and 'cloud_storage_type' in test_context.injected_args:
+            self.cloud_storage_type = test_context.injected_args[
```

Comment:
maybe assert that `cloud_storage_type` is one of the valid values?  Otherwise if someone typos the value, they'll get weird behavior.  Or you could have an `else` on the block below that throws.

Reply:
Done in https://github.com/redpanda-data/redpanda/compare/941c2cc83645b4fbcf6724407cdd3b477e6f3f1f..4d2353e885b4100a1fe7ce330c14d987238ad8d9.